sql/core/src/test/scala/org/apache/spark/sql/internal/CatalogSuite.scala

```diff
@@ -322,6 +325,14 @@ class CatalogSuite
     assert(e2.message == "Cannot create a file-based external data source table without path")
   }
 
+  test("dropTempView if a same-name table exists") {
+    withTable("same_name") {
+      sql("CREATE TABLE same_name(i int) USING json")
+      spark.catalog.dropTempView("same_name")
+      assert(spark.sessionState.catalog.tableExists(TableIdentifier("same_name")))
+    }
+  }
```

Comment:
Let's add comments to explain what this test is for in case we accidentally delete it in future.